backend/controllers/auth.js

```diff
@@ -1,20 +1,32 @@
 import { HttpError } from "../helpers/HttpError.js";
 import { createApiResponse } from "../helpers/createApiResponse.js";
 import { prisma } from "../lib/db.js";
+import jwt from "jsonwebtoken";
+import bcrypt from "bcrypt";
 
 const COOKIE_MAX_AGE_MILISECONDS = 60 * 60 * 1000;
+const { JWT_SECRET } = process.env;
 
 const register = async (req, res, next) => {
   try {
     const { username, password } = req.body;
+
+    const hashPassword = await bcrypt.hash(password, 10);
+
     const user = await prisma.user.create({
       data: {
-        password,
+        password: hashPassword,
         username,
       },
     });
+
+    const payload = {
+      id: user.id,
+    };
+    const token = jwt.sign(payload, JWT_SECRET, { expiresIn: "23h" });
+
     return res
-      .cookie("user", JSON.stringify({ username: user.username, id: user.id }), {
+      .cookie("token", token, {
         maxAge: COOKIE_MAX_AGE_MILISECONDS,
```

Comment:
set httpOnly setting on the cookie to prevent xss attacks from retrieving the cookie. This will make cookie inaccessible inside browser js, for example `document.cookie` will not be set in the browser.